src/Aspire.Hosting.Azure.Storage/AzureBlobStorageResource.cs

```diff
@@ -30,13 +30,20 @@ void IResourceWithAzureFunctionsConfig.ApplyAzureFunctionsConfiguration(IDiction
     {
         if (Parent.IsEmulator)
         {
-            target[connectionName] = Parent.GetEmulatorConnectionString();
+            var connectionString = Parent.GetBlobConnectionString();
+            target[connectionName] = connectionString;
+            // Injected to support Aspire client integration for Azure Storage.
+            target[$"{AzureStorageResource.BlobsConnectionKeyPrefix}__{connectionName}__ConnectionString"] = connectionString;
+            target[$"{AzureStorageResource.QueuesConnectionKeyPrefix}__{connectionName}__ConnectionString"] = connectionString;
```

Comment:
This is the "Blobs" resource. Why is it injecting Queues stuff?

Reply:
This is Functions specific. We're making the assumption that if a user is adding a reference to a Blob storage resource then they are probably using a BlobTrigger. Blob triggers take a dependency on queues to support retires across triggers (see [footnote 2 from this section of the Functions docs](https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-blob-trigger?tabs=python-v2%2Cisolated-process%2Cnodejs-v4%2Cextensionv5&pivots=programming-language-csharp#identity-based-connections) for more info.

I'll update the comment here.

Reply:
> Blob triggers take a dependency on queues to support retires across triggers

But the actual app (via Aspire client integrations) doesn't need to interact with a queue, do they? That's what line 37 is enabling.

Reply:
A, good point. We can probably remove the bits that are for the Aspire integration since we don't want to set up an expectation that you can access the underlying poison queue when you add a storage dependency.